zio/detector/reader.go

```diff
@@ -55,7 +55,7 @@ func NewReaderWithConfig(r io.Reader, zctx *resolver.Context, path string, cfg O
 
 	zngErr := match(zngio.NewReader(track, resolver.NewContext()), "zng")
```

Comment:
```suggestion
	zngErr := match(zngio.NewReaderWithOpts(track, resolver.NewContext(), zngio.ReaderOpts{Check: true}), "zng")
```

Reply:
I wondered about that one. On the whole I don't think type checking should be done in format detection - you can correctly identify a format even if a record doesn't typecheck. That said, since the other formats have type checking always-on, we may as well have it on here for consistency.